lib/chef/resource/yum_package.rb

```diff
@@ -38,7 +38,7 @@ class YumPackage < Chef::Resource::Package
       # its own arguments as unintended side-effects (and why functional languages that don't allow modifcations
       # of variables eliminate entire classes of bugs).
       # </lecture>
-      property :package_name, [ String, Array ], identity: true, coerce: proc { |x| x.is_a?(Array) ? x.to_a : x }
+      property :package_name, [ String, Array ], identity: true, coerce: proc { |x| x.is_a?(Array) ? x.to_a : ((x.strip.index(/\s/)) ? x.to_s.split(" ") : x) }
```

Comment:
A ternary does not seem to be needed for `x.is_a?(Array)` since the `true` case is doing a needless `to_a` against an object which should already be an array. 

Also, a ternary is probably the wrong thing to use for `x.strip.index(/\s/))`. Here is why, this condition will return true:

```ruby
[12] pry(main)> "one two three".strip.index(/\s/)
=> 3
[13] pry(main)> 0 ? "zero" : "not zero"
=> "zero"
[14] pry(main)> 1 ? "zero" : "not zero"
=> "zero"
```

Also, what is your intention with this: `x.to_s.split(" ")`? I ask because it can one of two things, one of which is effectively a no-op, and the other creates a complex nested array of strings. Eg:
```ruby
[16] pry(main)> %w(one two three).to_s.split(" ")
=> ["[\"one\",", "\"two\",", "\"three\"]"]
[17] pry(main)> "one two three".to_s.split(" ")
=> ["one", "two", "three"]
```

Reply:
I think that coercion is to convert an immutable array into a normal array, since apparently the provider may convert it into an immutable array.

Reply:
Also, looks like the intention with `x.to_s.split(" ")` was to convert the string into an array of strings.
```irb(main):024:0> y = '  '
=> "  "
irb(main):025:0> y.strip.index(/\s/) ? y.to_s.split(" ") : y
=> "  "
irb(main):026:0> y.strip.index(/\s/)
=> nil```

```irb(main):029:0> y='box bag bar  '
=> "box bag bar  "
irb(main):030:0> y.strip.index(/\s/) ? y.to_s.split(" ") : y
=> ["box", "bag", "bar"]```

If the user entered %w(one two three) the to_a true case would handle it and it would not result in the complex array.

But I admit, it does not seem to do the expected thing in the following scenarios with the above assumption:

``irb(main):034:0> y = 'foo'
=> "foo"
irb(main):035:0> y.strip.index(/\s/) ? y.to_s.split(" ") : y
=> "foo"
irb(main):036:0> y = 'foo '
=> "foo "
irb(main):037:0> y.strip.index(/\s/) ? y.to_s.split(" ") : y
=> "foo "
irb(main):038:0> y = 'foo  '
=> "foo  "
irb(main):039:0> y.strip.index(/\s/) ? y.to_s.split(" ") : y
=> "foo  "```